avocado/utils/archive.py

```diff
@@ -157,6 +163,27 @@ def extract(self, path='.'):
         :param path: destination path.
         """
         self._engine.extractall(path)
+        if self.is_zip:
+            self._update_symlinks(path)
+
+    def _update_symlinks(self, dst_dir='.'):
+        if platform.system() != "Linux":
+            LOG.warn("Symlink handling in zip files not supported on Windows.")
+            return
+        # Walk all files and re-create files as symlinks
+        for path, info in self._engine.NameToInfo.iteritems():
+            if info.external_attr & 2684354560 == 2684354560:
```

Comment:
I wonder if this magic number can be created with symbolic names.... and yes: `stat.S_IFLNK << 16` does the trick. 

Anyway, gives me the feeling that we may be exceeding what we should be supporting in Avocado.  Have you checked if there's any work on fixing this on Python itself?